packages/network-controller/src/NetworkController.ts

```diff
@@ -238,6 +240,10 @@ export class NetworkController extends BaseControllerV2<
         return this.ethQuery;
       },
     );
+
+    this.previousProvider = {
+      type: defaultState.providerConfig.type,
```

Comment:
It appears this is using the `type` from the default state, but that state can be overridden at construction time. What do you think about using:

``` suggestion
      type: this.state.providerConfig.type,
```

Reply:
done here https://github.com/MetaMask/core/pull/1132/commits/e3b0895dcc47c3ac0494016e49f87afdee468ef1